src/test/scala/org/broadinstitute/dsde/workbench/sam/service/TosServiceSpec.scala

```diff
@@ -29,82 +33,137 @@ class TosServiceSpec extends AnyFlatSpec with TestSupport with BeforeAndAfterAll
-    tosServiceEnabledV0.getTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe Option(false)
-    tosServiceEnabledV0.acceptTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe Option(true)
-    tosServiceEnabledV0.getTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe Option(true)
-
-    val tosServiceEnabledV2 = new TosService(dirDAO, "example.com", TestSupport.tosConfig.copy(enabled = true, version = "2"))
-    tosServiceEnabledV2.getTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe Option(false)
-    tosServiceEnabledV2.acceptTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe Option(true)
-    tosServiceEnabledV2.getTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe Option(true)
+    // reject and get ToS status
+    val tosServiceV2 = new TosService(dirDAO, TestSupport.tosConfig.copy(version = "0"))
+    val rejectTosStatusResult = tosServiceV2.rejectTosStatus(defaultUser.id, samRequestContext).unsafeRunSync()
+    assert(rejectTosStatusResult, s"rejectTosStatus(${defaultUser.id}) should reject the tos for the user")
+    val userRejectedToS = dirDAO.loadUser(defaultUser.id, samRequestContext).unsafeRunSync().orNull
+    val adherenceStatus = tosServiceV2.getTosAdherenceStatus(userRejectedToS).unsafeRunSync()
+    assertResult(expected = false, s"getTosStatus(${defaultUser.id}) should have returned false")(actual = adherenceStatus.userHasAcceptedLatestTos)
   }
 
-  "TosService.isTermsOfServiceStatusAcceptable" should "allow all requests to the API if TOS is disabled" in {
+  "TosService" should "allow users to accept new version of ToS" in {
     assume(databaseEnabled, databaseEnabledClue)
+
     dirDAO.createUser(defaultUser, samRequestContext).unsafeRunSync()
-    val tosServiceDisabled = new TosService(dirDAO, "example.com", TestSupport.tosConfig.copy(enabled = false))
-    tosServiceDisabled.isTermsOfServiceStatusAcceptable(defaultUser) should be(true)
+
+    // Accept V0
+    val tosServiceV0 = new TosService(dirDAO, TestSupport.tosConfig.copy(version = "0"))
+    tosServiceV0.getTosAdherenceStatus(defaultUser).unsafeRunSync() shouldBe TermsOfServiceAdherenceStatus(defaultUser.id, false, false)
+    tosServiceV0.acceptTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe true
+    val userAcceptedV0 = dirDAO.loadUser(defaultUser.id, samRequestContext).unsafeRunSync().orNull
+    val adherenceStatusV0 = tosServiceV0.getTosAdherenceStatus(userAcceptedV0).unsafeRunSync()
+    adherenceStatusV0.userHasAcceptedLatestTos shouldBe true
+
+    // Accept V2
+    val tosServiceV2 = new TosService(dirDAO, TestSupport.tosConfig.copy(version = "2"))
+    val adherenceStatusNotAcceptedLatest = tosServiceV2.getTosAdherenceStatus(userAcceptedV0).unsafeRunSync()
+    adherenceStatusNotAcceptedLatest.userHasAcceptedLatestTos shouldBe false
+    adherenceStatusNotAcceptedLatest.acceptedTosAllowsUsage shouldBe false
```

Comment:
For this test, I don't think I should be asserting on `.acceptedTosAllowsUsage`  Whether that is true or false is not part of what's under test here.  Yes, this coverage is more thorough or complete, but it adds complexity to read and maintain and introduces ways for this test to fail that are not the intended focus per the name of the test.  Surely we have other tests that assert all the various permutations and calculations for `.acceptedTosAllowsUsage` and those should cover the scenario constructed here and should do so more thoroughly and precisely, thus negating the need for the additional/duplicative testing here.